py/client/tests/test_query.py

```diff
@@ -64,6 +64,28 @@ def test_update_by(self):
         result_table = query.exec()
         self.assertTrue(result_table.size == 10)
 
+    def test_snapshot(self):
+        test_table = self.session.time_table(period=1000000)
+        while test_table.snapshot().size < 100:
+            time.sleep(0.001)
```

Comment:
Are there any cases where we could accidentally hang forever?

Reply:
Not likely. Even when the server somehow falls into a dead-lock situation, the gRPC/network layer should eventually time out and exception out.